Preflight Checks (#116)
itential/itential.deployer: docs/preflight_guide.md

# Overview

> The playbook and role in this section are designed to run preflight checks on the hosts for Redis, Mongodb, Platform, and Gateway. These checks will determine if the host meets the minimum recomended system specifications for the given application.

Reviewer: ```suggestion
The playbook and role in this section are designed to run preflight checks on the hosts for Redis, Mongodb, Platform, and Gateway. These checks will determine if the host meets the minimum recommended system specifications for the given application.
```

# Preflight Role

## General information

> This role can be run seperatly to check the inventory host as well as when running the installtion role for the other applications. By default, when installing an application using the deployer, the preflight role will run however, it will ignore the results so the application will still be installed. This behaviour is controlled with the variables `run_preflight` and `ignore_preflight_checks`. When `ignore_preflight_checks` is set to false, and any of the preflight checks fail, installition on the application will not preceed.

Reviewer: ```suggestion
This role can be run separately to check the inventory host as well as when running the installation role for the other applications. By default, when installing an application using the deployer, the preflight role will run however, it will ignore the results so the application will still be installed. This behavior is controlled with the variables `run_preflight` and `ignore_preflight_checks`. When `ignore_preflight_checks` is set to false, and any of the preflight checks fail, installition on the application will not preceed.
```

Reviewer: Are you British? :)

> This role can be run separately to check the inventory host as well as when running the installation role for the other applications. By default, when installing an application using the deployer, the preflight role will not run however, it will ignore the results so the application will still be installed. This behavior is controlled with the variables `run_preflight` and `ignore_preflight_checks`. When `ignore_preflight_checks` is set to false, and any of the preflight checks fail, installation on the application will not proceed.

Reviewer: ```suggestion
This role can be run separately to check the inventory host as well as when running the installation role for the other applications. By default, when installing an application using the deployer, the preflight role will not run. This behavior is controlled with the variables `preflight_run_checks` and `preflight_enforce_checks`. When `preflight_enforce_checks` is set to true, and any of the preflight checks fail, installation on the application will not proceed.
```

The following checks will be made against the host. 

| Check | Description | Cause Failure
| :---- | :---------- | :---------- 
| `OS`  | What version of Redhat/Rocky is installed on the host | Yes
| `CPU` | How many CPU cores the host has | Yes
| `RAM` | How many RAM the host has | Yes
| `Disk Space` | How much free disk space the host has | Yes
| `SELinux` | Is SELinxus being enforced | No
| `IPv6` | Is IPv6 configured | No
| `HTTP Proxy` | Is there an HTTP proxy | No
| `HTTPS Proxy` | Is there an HTTPS proxy | No
| `Ports` | Are ports needed for the application blocked by firewalls | No
| `URLs` | Does the host have access to required URLs | No
| `AVX` | Is AVX supported (MongoDB only) | No

> The preflight role will run checks against the hosts and then trasfer the results into a local directory defined by the `preflight_directory` for review. 

Reviewer: ```suggestion
The preflight role will run checks against the hosts and then transfer the results into a local directory defined by the `preflight_directory` for review. 
```
